pkg/proxy/iptables/proxier.go

```diff
@@ -807,12 +795,11 @@ func (proxier *Proxier) execConntrackTool(parameters ...string) error {
 // The only other iptables rules are those that are setup in iptablesInit()
 // assumes proxier.mu is held
 func (proxier *Proxier) syncProxyRules() {
-	if proxier.throttle != nil {
-		proxier.throttle.Accept()
+	if !proxier.throttle.allowSync() {
+		return
```

Comment:
Hold up.  syncProxyRules() is called by `OnEndpointsUpdate` and `OnServiceUpdate`, both of which call something AFTER the sync.  They expect that the sync actually sync'ed, and if it didn't (was deferred) this is going to do the wrong thing.

I think we need a larger change than this, I am sad to say.  We need to defer 

I am sorry this review is so painful - this code is very async and overly delicate.